### PR TITLE
bump: operator-rs. Switch to workspace dependencies.

### PR DESCRIPTION
**Pull request by sbernauer**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -3,5 +3,28 @@ members = [
...
+serde_json = "1.0"
+serde_yaml = "0.9"
+snafu = "0.7"
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.45.1" }
+strum = { version = "0.25", features = ["derive"] }
+tokio = { version = "1.29", features = ["full"] }
+tracing = "0.1"
+
+[patch."https://github.com/stackabletech/operator-rs.git"]
+stackable-operator = { git = "https://github.com/stackabletech//operator-rs.git", branch = "bump/kube" }
```
Comment (sbernauer) [inline Cargo.toml:30]: FIXME: Switch to 0.46.0 once released

File: rust/crd/Cargo.toml [hunk 2]
```diff
@@ -5,20 +5,19 @@ edition = "2021"
 license = "OSL-3.0"
 name = "stackable-trino-crd"
 repository = "https://github.com/stackabletech/trino-operator"
-version = "0.0.0-dev"
 publish = false
+version.workspace = true
```
Comment (sbernauer) [inline rust/crd/Cargo.toml:9]: This will probably break release scripts CC @adwk67 @razvan

File: rust/operator-binary/Cargo.toml [hunk 3]
```diff
@@ -6,27 +6,25 @@ edition = "2021"
...
+stackable-operator = { workspace = true }
+strum = { workspace = true }
+tokio = { workspace = true }
+tracing = { workspace = true }
 
 [build-dependencies]
-built = { version =  "0.6", features = ["chrono", "git2"] }
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.45.1" }
-stackable-trino-crd = { path = "../crd" }
+built = { workspace = true }
```
Comment (Techassi) [inline rust/operator-binary/Cargo.toml:30]: Is it intentional that you removed these two features? If yes, ignore this comment.

Comment (Techassi) [reply rust/operator-binary/Cargo.toml:30]: Well never mind. They are now enabled workspace-wide...

File: Cargo.toml [hunk 4]
```diff
@@ -3,5 +3,28 @@ members = [
     "rust/crd", "rust/operator-binary"
```
Comment (Techassi) [inline Cargo.toml:3]: We could shorten this to `"rust/*"`

File: rust/crd/Cargo.toml (hunk 2 above)
Comment (razvan) [reply rust/crd/Cargo.toml:9]: This is fine. The release scripts don't change dependencies.

File: Cargo.toml (hunk 4 above)
Comment (razvan) [reply Cargo.toml:3]: This will break `cargo-version.py` release script. Can be fixed, no bid deal, just FYI.

File: rust/operator-binary/Cargo.toml (hunk 3 above)
Comment (sbernauer) [reply rust/operator-binary/Cargo.toml:30]: exactly :)

File: Cargo.toml (hunk 4 above)
Comment (sbernauer) [reply Cargo.toml:3]: @razvan I would leave the final decision to you

File: rust/crd/Cargo.toml (hunk 2 above)
Comment (sbernauer) [reply rust/crd/Cargo.toml:9]: Looking at https://github.com/stackabletech/stackable-utils/blob/main/release/cargo-version.py are you sure it works? ;) Just want to be save

**Comment by sbernauer** [comment]
Started https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/trino-operator-it-custom/92/

File: rust/crd/Cargo.toml (hunk 2 above)
Comment (sbernauer) [reply rust/crd/Cargo.toml:9]: I think I would suggest to revert the version consolidation change as it does not seem worth it to me. WDYT?
```
➜  trino-operator git:(bump/kube) /home/sbernauer/stackabletech/stackable-utils/release/cargo-version.py --set 0.0.0-pr123
Traceback (most recent call last):
  File "/home/sbernauer/stackabletech/stackable-utils/release/cargo-version.py", line 172, in <module>
    new.save(old)
  File "/home/sbernauer/stackabletech/stackable-utils/release/cargo-version.py", line 138, in save
    self.crates[cn].save(previous.crates[cn])
  File "/home/sbernauer/stackabletech/stackable-utils/release/cargo-version.py", line 83, in save
    line = line.replace(previous.version, self.version)
TypeError: replace() argument 1 must be str, not dict
```

File: Cargo.toml (hunk 4 above)
Comment (razvan) [reply Cargo.toml:3]: I would merge this and replace `cargo-version.py` with [cargo set-version](https://lib.rs/crates/cargo-edit) in the release scripts. We already use it in the GH actions.

File: rust/crd/Cargo.toml (hunk 2 above)
Comment (sbernauer) [reply rust/crd/Cargo.toml:9]: Ah, awesome! https://github.com/stackabletech/trino-operator/blob/f0243a415e1d25c7cf60a2da55cf61a8d53f237d/.github/workflows/build.yml#L317
I can confirm `cargo set-version --offline --workspace 0.0.0-pr123` still works :+1:

**Comment by sbernauer** [comment]
@razvan please have a final look and approve if it looks good, so that we can continue with the other operators

File: tests/templates/kuttl/authentication/create-trino-cluster.yaml.j2 [hunk 5]
```diff
@@ -17,7 +17,7 @@ metadata:
   namespace: $NAMESPACE
 spec:
   image:
-    productVersion: "{{ test_scenario['values']['trino-latest'] }}"
+    productVersion: "'"{{ test_scenario['values']['trino-latest'] }}"'"
```
Comment (razvan) [inline tests/templates/kuttl/authentication/create-trino-cluster.yaml.j2:20]: This generates fields like this:
```
productVersion: "'"414"'" # double-quote - quote - double-quote 4 1 4 ...
```
Is this intentional ?

**Review by razvan — COMMENTED** [review]
One question

File: tests/templates/kuttl/authentication/create-trino-cluster.yaml.j2 (hunk 5 above)
Comment (sbernauer) [reply tests/templates/kuttl/authentication/create-trino-cluster.yaml.j2:20]: We actually need this change to fix the kuttl nightly tests. As this runs through an "echo" statement we need to quote it twice, as bash removes the outer quotes

**Review by razvan — APPROVED** [review]

**Comment by sbernauer** [comment]
Thanks for review!
